### PR TITLE
Use Elasticsearch cluster health for check instead of ping

### PR DESCRIPTION
**Pull request by inkel**
As stated in #287, using ping will only allow returning healthy or unhealthy statuses. By using
[Elasticsearch Cluster Health API](https://www.elastic.co/guide/en/elasticsearch/reference/current/cluster-health.html) we can also return degraded if the cluster is in the yellow state.

While implementing this I realized that for some use cases users won't need to check for degraded, so to keep backwards dependency I've added a new option method `ElasticsearchOptions` to enable using cluster health check:

```csharp
var options = new ElasticsearchOptions()
    .UseClusterHealth();
```

By default, it will use the original ping check.

### CODE REVIEW

**Comment by Xabaril** [comment]
Hi @inkel 

Let me some days to review this PR

**Comment by inkel** [comment]
@unaizorrilla I've added the tests without needing to spawn Elasticsearch demons but rather mocking an HTTP server that returns the same payload as Cluster Health API. Please check commit 9220378 and let me know what you think.

Nevertheless, I think this is ready for review!

**Comment by inkel** [comment]
@Xabaril hi! I've just pushed some updates with tests for the new functionality. Yes, please, take as long as you need! And thank you for publishing all these packages, we are using it at my company and they are wonderful, they just work 😸 

I do have more PRs in mind, though, I hope you don't mind 😬

**Comment by inkel** [comment]
🤔 AppVeyor tests are failing due to not finding a right .NET framework:

```
dotnet.exe : Testhost process exited with error: It was not possible to find any compatible framework version
At C:\projects\aspnetcore-diagnostics-healthchecks\build.ps1:46 char:16
+         exec { & dotnet test}
+                ~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (Testhost proces...amework version:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
 
The specified framework 'Microsoft.AspNetCore.App', version '3.0.0' was not found.
  - Check application dependencies and target a framework version installe
d at:
      C:\Program Files\dotnet
  - Installing .NET Core prerequisites might help resolve this problem:
      https://go.microsoft.com/fwlink/?LinkID=798306&clcid=0x409
  - The .NET Core framework and SDK can be installed from:
      https://aka.ms/dotnet-download
  - The following versions are installed:
      2.1.0 at [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
      2.1.1 at [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
      2.1.2 at [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
      2.1.3 at [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
      2.1.4 at [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
      2.1.5 at [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
      2.1.6 at [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
      2.1.7 at [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
      2.1.9 at [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
      2.1.10 at [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
      2.1.11 at [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
      2.1.12 at [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
      2.2.0 at [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
      2.2.1 at [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
      2.2.4 at [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
      2.2.5 at [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
      2.2.6 at [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
. Please check the diagnostic logs for more information.
Test Run Aborted.
```

**Comment by unaizorrilla** [comment]
Let me check appveyor because is now failing and we don't change anything, the same build that works some days ago is failing now! 👎

**Comment by unaizorrilla** [comment]
Hi @inkel 

I think the issue is related with https://github.com/microsoft/vstest/issues/2189 , I'm trying to solve it asap!

**Comment by unaizorrilla** [comment]
I think there is a bug on Microsoft.Net.Test.Sdk 16.3.0 , downgrade to 16.2.0 build is working again! I try to review PR asap

**Comment by inkel** [comment]
@unaizorrilla any comments on this one?

**Comment by unaizorrilla** [comment]
Hi @inkel 

we try to review and merge it asap

File: src/HealthChecks.Elasticsearch/ElasticsearchOptions.cs [hunk 1]
```diff
@@ -12,6 +12,7 @@ public class ElasticsearchOptions
         public X509Certificate Certificate { get; private set; }
         public bool AuthenticateWithBasicCredentials { get; private set; } = false;
         public bool AuthenticateWithCertificate { get; private set; } = false;
+        public bool CheckClusterHealth { get; private set; } = false;
```
Comment (sungam3r) [inline src/HealthChecks.Elasticsearch/ElasticsearchOptions.cs]:
```suggestion
        public bool CheckClusterHealth { get; private set; }
```

**Comment by inkel** [comment]
Thanks @sungam3r for the suggestion!

**Comment by inkel** [comment]
Hi, are there any updates on this?

**Comment by inkel** [comment]
Hi! Sorry for being pushy, but is there any update on this one?

**Comment by sungam3r** [comment]
@inkel Hi. I help to maintain this repo. I found this PR looking through open issues (#287). If you want to get this merged then I can review. Reopen and rebase onto master if you interested.

**Comment by inkel** [comment]
Hi @sungam3r! I'm not working with # anymore at the moment, and I've even left the company I was working at the time we needed this merged. I'd love to see the changes integrated as I think it's the right way to have a health check on Elasticsearch, but I cannot make any changes to this branch nor help maintaining it.

**Comment by sungam3r** [comment]
Covered by #1577 .

**Comment by inkel** [comment]
Oh wow, what a blast from the past seeing action in this PR.

I'm glad this is now covered.

**Pull request closed without merging**